src/main/java/io/gravitee/policy/javascript/JavascriptPolicy.java

```diff
@@ -15,319 +15,230 @@
-                .getComponent(Vertx.class)
-                .executeBlocking(
-                    event -> {
-                        try {
-                            eval(script, scriptContext);
-                            event.complete();
-                        } catch (Exception e) {
-                            event.fail(e);
-                        }
-                    },
-                    event -> {
-                        if (event.failed()) {
-                            logger.error("Unable to run Javascript script", event.cause());
-                            policyChain.failWith(io.gravitee.policy.api.PolicyResult.failure(event.cause().getMessage()));
-                        } else {
-                            PolicyResult result = (PolicyResult) scriptContext.getAttribute(RESULT_VARIABLE_NAME);
-
-                            if (result.getState() == PolicyResult.State.SUCCESS) {
-                                policyChain.doNext(request, response);
-                            } else {
-                                if (result.getContentType() != null) {
-                                    policyChain.failWith(
-                                        io.gravitee.policy.api.PolicyResult.failure(
-                                            result.getKey(),
-                                            result.getCode(),
-                                            result.getError(),
-                                            result.getContentType()
-                                        )
-                                    );
-                                } else {
-                                    policyChain.failWith(
-                                        io.gravitee.policy.api.PolicyResult.failure(result.getKey(), result.getCode(), result.getError())
-                                    );
-                                }
-                            }
-                        }
-                    }
-                );
         }
 
-        return null;
+        return Completable.complete();
     }
 
-    private String executeStreamScript(JsRequest request, JsResponse response, JsExecutionContext executionContext, String script)
-        throws PolicyFailureException {
-        // Prepare binding
-        final ScriptContext scriptContext = createScriptContext(request, response, executionContext);
-        final PolicyResult result = (PolicyResult) scriptContext.getAttribute(RESULT_VARIABLE_NAME);
+    private Maybe<Buffer> onHttpContent(
+        HttpPlainExecutionContext ctx,
+        Buffer bodyBuffer,
+        HttpHeaders headers,
+        ScriptContext scriptContext
+    ) {
+        return scriptFlowable
+            .concatMapMaybe(script -> runContentAwareScript(ctx, scriptContext, script))
+            .lastElement()
+            .filter(jsBuffer -> configuration.isOverrideContent())
+            .doOnSuccess(jsBuffer -> setContentLength(headers, jsBuffer))
```

Comment:
We should use https://github.com/gravitee-io/gravitee-gateway-api/blob/master/src/main/java/io/gravitee/gateway/reactive/api/context/http/HttpPlainResponse.java#L155 to set the content length

```suggestion
            .doOnSuccess(jsBuffer -> ctx.response().contentLength(buffer.length()))
```

Reply:
I've used ctx.request().contentLength / ctx.response().contentLength depending on the phase.